src/cheri/insns/jalr_32bit.adoc

```diff
@@ -14,22 +14,47 @@ include::wavedrom/ct-unconditional-2.adoc[]
 
 include::base_isa_extension.adoc[]
 
+#Begin changed since last ARC review#
+
 Description::
 Indirect jump to the target capability in `{cs1}` with an address offset.
 +
 Copy `{cs1}` to the target <<pcc>>
 +
-. If the target <<pcc>> is a <<sentry_cap>> and the `offset` is zero, then unseal the target <<pcc>>.
-Otherwise, increment the address of the target <<pcc>> by the sign-extended 12-bit `offset`.
+. Increment the address of the target <<pcc>> by the sign-extended 12-bit `offset`.
+. If the target <<pcc>> is a <<sentry_cap>>, then
+  - clear the {ctag} of the target <<pcc>> if the triple of its <<sec_cap_type>> value, `{cd}`, and `{cs1}` is not acceptable, as defined by the capability encoding, and
+  - and unseal the target <<pcc>>.
 . Set bit zero of the target address to zero and update <<pcc>> with the new target address using the semantics of the <<SCADDR>> instruction
-. Jump to the target <<pcc>>
+. If `{cd}` is not `null`, add the width of this instruction to the current <<pcc>> and store the result into `{cd}`.
+. If the capability encoding defines <<sentry_cap>> <<sec_cap_type>> values for return pointers, seal the capability in `{cd}` as specified.
+. Jump to the target <<pcc>>.
+. Apply any capability encoding specified architectural semantics for the type of `{cs1}`.
 +
-The <<pcc>> of the next instruction is calculated using the semantics of the <<SCADDR>> instruction, and written to `{cd}` as a <<sentry_cap>>.
+The <<pcc>> of the next instruction is calculated using the semantics of the <<SCADDR>> instruction.
 +
 NOTE: Because of the unsealing and <<SCADDR>> behavior, it is not possible for the target <<pcc>> to be sealed.
 +
 NOTE: A future extension may raise an exception on the JALR instruction itself if the target <<pcc>> will raise a CHERI exception at the target.
 
+[NOTE]
+=====
+<<JALR_CHERI>> defines three hooks for use by the capability encoding:
+
+1. Architectural interpretation of zero or more particular <<sec_cap_type>> values when a capability is passed as the input to <<JALR_CHERI>>.
+   These <<sec_cap_type>> values are, by definiton, <<sentry_cap>> types.
+
+2. A subset of those <<sec_cap_type>> values for use by <<JALR_CHERI>> to seal the return capability installed in the link register.
+   A sentry return capability grants the right to return exactly to the instruction after the caller's call without relative offset.
+   Moreover, a sentry return capability does not grant the callee access to the memory pointed to by the caller's <<pcc>>.
+
+3. Conditional behavior of <<JALR_CHERI>> based on the _register selectors_ (`{cd}`, `{cs1}`) used in an instance of the instruction together with the <<sec_cap_type>> value of the capability in `{cs1}`.
+
+In all cases, when a sealed capability is used as the input to <<JALR_CHERI>>, the instruction must have a zero offset or the target <<pcc>> will become unsealed, since the `offset` is interpreted with <<SCADDR>> semantics.
```

Comment:
```suggestion
In all cases, when a sealed capability is used as the input to <<JALR_CHERI>>, the instruction must have a zero offset or the target <<pcc>> will become untagged, since the `offset` is interpreted with <<SCADDR>> semantics.
```

I believe this is a typo?

Reply:
I changed it from `untagged` as I've purged that word from the spec